internal/pkg/deploy/cloudformation/pipeline.go

```diff
@@ -34,7 +36,28 @@ func (cf CloudFormation) CreatePipeline(in *deploy.CreatePipelineInput) error {
 	if err != nil {
 		return err
 	}
-	return cf.cfnClient.CreateAndWait(s)
+	err = cf.cfnClient.CreateAndWait(s)
+	if err != nil {
+		return err
+	}
+
+	output, err := cf.cfnClient.Outputs(s)
+	if err != nil {
+		return err
+	}
+	// If the pipeline has a PipelineConnectionARN, indicating that it is has a CodeStarConnections source provider, the user needs to update the connection status; Copilot will wait until that happens.
+	if output["PipelineConnection"] != "" {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(45*time.Minute))
```

Comment:
Can we convert this to an early return pattern? ie

```suggestion
	if output["PipelineConnection"] == "" {
	        return nil
	}
	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(45*time.Minute))
```

Reply:
Ah, yes. Thanks!